src/resmom/mom_main.c

```diff
@@ -1306,7 +1306,8 @@ initialize(void)
 	 *	Check that there are no bad combinations of sharing values
 	 *	across the vnodes.
 	 */
-	avl_create_index(&ix, AVL_NO_DUP_KEYS, 0);
+	if (avl_create_index(&ix, AVL_NO_DUP_KEYS, 0))
+		exit(1);
```

Comment:
I think you need to do a die(0) here so as to cleanup stuff,.. like in a later part of the code.